warp10/src/main/java/io/warp10/script/functions/GET.java

```diff
@@ -47,42 +47,74 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     
     if (coll instanceof Map) {
       value = ((Map) coll).get(key);
-    } else {
-      if (!(key instanceof Number)) {
-        throw new WarpScriptException(getName() + " expects the key to be an integer when operating on a list.");
-      }
+
+    } else if (key instanceof Number) {
       int idx = ((Number) key).intValue();
 
       if (coll instanceof List) {
         int size = ((List) coll).size();
 
-        idx = computeAndCheckIndex(this, idx, size);
+        idx = computeAndCheckIndex(idx, size);
 
         value = ((List) coll).get(idx);
+
       } else {
         int size = ((byte[]) coll).length;
 
-        idx = computeAndCheckIndex(this, idx, size);
+        idx = computeAndCheckIndex(idx, size);
 
         value = (long) (((byte[]) coll)[idx] & 0xFFL);
       }
+
+    } else if (coll instanceof byte[]) {
+      throw new WarpScriptException(getName() + " expects the key to be an integer when operating on a byte array.");
+
+    } else if (!(key instanceof List)) {
+      throw new WarpScriptException(getName() + " expects the key to be an integer or a list of integers when operating on a List.");
+
+    } else {
+      for (Object o: (List) key) {
+        if (!(o instanceof Number)) {
+          throw new WarpScriptException(getName() + " expects the key to be an integer or a list of integers when operating on a List.");
```

Comment:
A list of Double will get through here, should we check that elements are Longs which is the only integer type generated by WarpScript?

Reply:
I don't think so if we want to be coherent with GET e.g. "[ 0 ] 0.3 GET" doesn't raise an exception and gives 0

Reply:
Following our discussion, should also change behavior of GET on Doubles